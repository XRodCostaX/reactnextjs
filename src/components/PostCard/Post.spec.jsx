import { render, screen } from '@testing-library/react';
import { describe } from 'vitest';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;
describe('<PostCard />', () => {
  it('Should render PostCard correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: /title 1/i })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: /title 1/i })).toBeInTheDocument();

    expect(screen.getByText('body 1')).toBeInTheDocument();
  });

  it('should match a snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
