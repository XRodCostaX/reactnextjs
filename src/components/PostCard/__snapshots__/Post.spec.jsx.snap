// Vitest Snapshot v1

exports[`<PostCard /> > should match a snapshot 1`] = `
<div
  class="sc-bcXHqe kJyWoP"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 1
        1
      </h2>
      <p>
        body 1
      </p>
    </div>
  </div>
</div>
`;
